src/gui/qgsqueryresultwidget.cpp

```diff
@@ -0,0 +1,493 @@
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsqueryresultwidget.h"
+#include "qgsabstractdatabaseproviderconnection.h"
+#include "qgscodeeditorsql.h"
+#include "qgsmessagelog.h"
+#include "qgsquerybuilder.h"
+#include "qgsvectorlayer.h"
+
+QgsQueryResultWidget::QgsQueryResultWidget( QWidget *parent, QgsAbstractDatabaseProviderConnection *connection )
+  : QWidget( parent )
+{
+  setupUi( this );
+
+  // Unsure :/
+  // mSqlEditor->setLineNumbersVisible( true );
+
+  mQueryResultsTableView->hide();
+  mQueryResultsTableView->setItemDelegate( new QgsQueryResultItemDelegate( mQueryResultsTableView ) );
+  mProgressBar->hide();
+
+  connect( mExecuteButton, &QPushButton::pressed, this, &QgsQueryResultWidget::executeQuery );
+  connect( mClearButton, &QPushButton::pressed, this, [ = ]
+  {
+    mSqlEditor->setText( QString() );
+  } );
+  connect( mLoadLayerPushButton, &QPushButton::pressed, this, [ = ]
+  {
+    if ( mConnection )
+    {
+      emit createSqlVectorLayer( mConnection->providerKey(), mConnection->uri(), sqlVectorLayerOptions() );
+    }
+  }
+         );
+  connect( mSqlEditor, &QgsCodeEditorSQL::textChanged, this, [ = ] { updateButtons(); } );
+  connect( mFilterToolButton, &QToolButton::pressed, this, [ = ]
+  {
+    if ( mConnection )
+    {
+      try
+      {
+        std::unique_ptr<QgsVectorLayer> vlayer { mConnection->createSqlVectorLayer( sqlVectorLayerOptions() ) };
+        QgsQueryBuilder builder{ vlayer.get() };
+        if ( builder.exec() == QDialog::Accepted )
+        {
+          mFilterLineEdit->setText( builder.sql() );
+        }
+      }
+      catch ( const QgsProviderConnectionException &ex )
+      {
+        mMessageBar->pushCritical( tr( "Error opening filter dialog" ), tr( "There was an error while preparing SQL filter dialog: %1." ).arg( ex.what() ) );
+      }
+    }
+  } );
+
+  // Sync all SQL layer options
+  connect( mSqlEditor, &QgsCodeEditorSQL::textChanged, this, &QgsQueryResultWidget::syncSqlOptions );
+  connect( mFilterLineEdit, &QLineEdit::textChanged, this, &QgsQueryResultWidget::syncSqlOptions );
+  connect( mPkColumnsComboBox, &QgsCheckableComboBox::checkedItemsChanged, this, [ = ]( const QStringList & ) { syncSqlOptions(); } );
+  connect( mGeometryColumnComboBox, &QComboBox::currentTextChanged, this, [ = ]( const QString & ) { syncSqlOptions(); } );
```

Comment:
Make syncSqlOptions a private slot?

Reply:
It is already, here the lambda is required to ignore the argument from the sender.